src/utils/memory_size.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <multipass/exceptions/invalid_memory_size_exception.h>
+#include <multipass/memory_size.h>
+#include <multipass/utils.h>
+
+#include <fmt/format.h>
+
+namespace mp = multipass;
+
+namespace
+{
+constexpr auto kilo = 1024LL;
+constexpr auto mega = kilo * kilo;
+constexpr auto giga = mega * kilo;
+
+long long in_bytes(const std::string& mem_value)
+{
+    QRegExp matcher("^(\\d+)([KMG])?B?$", Qt::CaseInsensitive);
+
+    if (matcher.exactMatch(QString::fromStdString(mem_value)))
```

Comment:
If using `exactMatch()`, you can omit the `^` and `$` in the expression above.

Reply:
Good point, will do that (to make the regex simpler to read).